src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -39,18 +40,83 @@ public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TVal
                 return false;
             }
 
-            if (s_isUint64)
-            {
-                if (reader.TryGetUInt64(out ulong ulongValue))
-                {
-                    value = (TValue)Enum.ToObject(valueType, ulongValue);
-                    return true;
-                }
-            }
-            else if (reader.TryGetInt64(out long longValue))
+            // When utf8reader/writer will support all primitive types we should remove custom bound checks
+            // https://github.com/dotnet/corefx/issues/36125
+
+            switch (s_enumTypeCode)
             {
-                value = (TValue)Enum.ToObject(valueType, longValue);
-                return true;
+                case TypeCode.SByte:
+                    {
+                        if (reader.TryGetInt32(out int byte8) && JsonHelpers.IsInRangeInclusive(byte8, sbyte.MinValue, sbyte.MaxValue))
+                        {
+                            value = Unsafe.As<int, TValue>(ref byte8);
```

Comment:
This should be:

```
{
    byte byteValue = (byte)byteValue;
    value = Unsafe.As<byte, TValue>(ref );
}
 ```

Reply:
ah ok(`Right, you need to do a regular cast first to the right size, and only use Unsafe.As on the local of the right size.`)

Reply:
Btw not so clear why As(int,TValue) with TValue byte it's not correct...if you've some time can you elaborate?First time with this unsafe helpers.

Reply:
It does not work on big endian machines. Big endian machines store the most significant byte first.

.NET Core does not support any big endian machines today, but Mono that is reusing this code does. We try to make sure that all code that we write is big endian friendly.

Reply:
ok thank's...PTAL @JeremyKuhne